src/network/networkd-dhcp4.c

```diff
@@ -1499,6 +1499,12 @@ static int dhcp4_configure(Link *link) {
                         return log_link_debug_errno(link, r, "DHCPv4 CLIENT: Failed to set IP service type: %m");
         }
 
+        if (link->network->dhcp_socket_priority >= 0) {
```

Comment:
hmm, are you sure only positive priorities are allowed? The way i read the kernel code no range check is done and negative priorities are ok too?

Reply:
Yes you're right. Usual values are 0-6 but some people may use it for very specific stuff.
I removed the check.